src/elm/Page/Community/ObjectiveEditor.elm

```diff
@@ -411,6 +423,38 @@ update msg model loggedIn =
                     |> UR.addExt
                         (Just ClickedSaveObjective |> RequiredAuthentication)
 
+        ClickedArchiveObjetive ->
+            case model.objectiveId of
+                Just objectiveId ->
+                    model
+                        |> UR.init
+                        |> UR.addCmd
+                            (Api.Graphql.mutation loggedIn.shared
+                                (Mutation.completeObjective
+                                    { input = { objectiveId = objectiveId } }
+                                    empty
+                                )
+                                CompletedArchiveMutation
+                            )
+
+                Nothing ->
+                    model
+                        |> UR.init
+
+        CompletedArchiveMutation (Ok _) ->
+            let
+                newModel =
+                    { model | archived = True }
+            in
+            newModel
+                |> UR.init
```

Comment:
we should redirect it to `community/${community}/objectives` before showing the feedback